File: 오선민/programmers/P181860.py
'''
문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/181860
시간: 0.04ms
메모리: 10.2MB
PR 제목 : 날짜 사이트_명 문제_번호 사용_언어
'''

def solution(arr, flag):
    x = []
    for idx, f in enumerate(flag):
        if f:
            x.extend([arr[idx]] * 2 * arr[idx])
        else:
            del x[-arr[idx]:]
    return x

File: 공예진/programmers/P181860.java
//문제: https://school.programmers.co.kr/learn/courses/30/lessons/181860
//시간: 0.40ms
//메모리: 83.6MB

import java.util.*;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < flag.length; i++) {
            if (flag[i]) {
                for (int j = 0; j < arr[i] * 2; j++) {
                    list.add(arr[i]);
                }
            } else {
                for (int j = 0; j < arr[i]; j++) {
                    list.remove(list.size() - 1);
                }
            }
        }

        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }

        return answer;
    }
}